cmd/symmecrypt/main.go

```diff
@@ -0,0 +1,95 @@
+package main
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/howeyc/gopass"
+	"github.com/ovh/symmecrypt"
+	"github.com/ovh/symmecrypt/ciphers/aesgcm"
+	"github.com/ovh/symmecrypt/ciphers/aespmacsiv"
+	"github.com/ovh/symmecrypt/ciphers/chacha20poly1305"
+	"github.com/ovh/symmecrypt/ciphers/hmac"
+	"github.com/ovh/symmecrypt/keyloader"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var (
+	DefaultCipher = aesgcm.CipherName
+	ciphers       = []string{aesgcm.CipherName, chacha20poly1305.CipherName, aespmacsiv.CipherName, hmac.CipherName}
+
+	app = kingpin.New("symmecrypt CLI tool", fmt.Sprintf("A command-line utility to generate symmecrypt encryption keys. Available ciphers: %s (default: %s)", strings.Join(ciphers, ", "), DefaultCipher))
+
+	newEncryption           = app.Command("new", "Generate a new random encryption key")
+	newEncryptionIdentifier = newEncryption.Arg("identifier", "identifier for the key").Required().String()
+	newEncryptionCipher     = newEncryption.Arg("cipher", "cipher for the key").Default(DefaultCipher).Enum(ciphers...)
+
+	encrypt       = app.Command("encrypt", "Encrypt data with selected key + cipher. Outputs hex.")
+	encryptCipher = encrypt.Arg("cipher", "cipher to use").Default(DefaultCipher).Enum(ciphers...)
+
+	decrypt       = app.Command("decrypt", "Decrypt data with selected key + cipher. Input is hex.")
+	decryptCipher = decrypt.Arg("cipher", "cipher to use").Default(DefaultCipher).Enum(ciphers...)
+)
+
+func main() {
+
+	cmd := kingpin.MustParse(app.Parse(os.Args[1:]))
+
+	switch cmd {
+	case newEncryption.FullCommand():
+		key, err := keyloader.GenerateKey(*newEncryptionCipher, *newEncryptionIdentifier, false, time.Now())
+		if err != nil {
+			panic(err)
```

Comment:
panic is not very user-friendly

why not print a nice error :
```go
fmt.Printf("error: unable to generate key: %v\n", err)
os.Exit(1)
```

or 
```go
log.Fatalf("error: unable to generate key: %v", err) 
```

or

define a `checkErr` function
```go
func checkErr(err error, format string, args ....interface{}) {
    if err == nil {
        return
    }
    msg := fmt.Sprintf(format, args...)
    fmt.Printf(msg + ": %v\n", err)
    os.Exit(1)
}
```

and replace all 
```go
if err != nil {
    panic(err)
}
```
by 
```go
checkErr(err, "my pretty error message")
```